tests/phpunit/unit/api/compiler/beans-page-compiler/compilePageStyles.php

```diff
@@ -0,0 +1,406 @@
+		Monkey\Functions\when( 'get_option' )->justReturn( true );
+		Monkey\Functions\expect( '_beans_is_compiler_dev_mode' )->once()->andReturn( true );
+		Monkey\Functions\expect( 'beans_get' )->never();
+		Monkey\Functions\expect( 'beans_compile_css_fragments' )->never();
+
+		// Run the tests.
+		$this->assertNull( ( new _Beans_Page_Compiler() )->compile_page_styles() );
+	}
+
+	/**
+	 * Test _Beans_Page_Compiler::compile_page_styles() should not compile when there are not styles.
+	 */
+	public function test_should_not_compile_when_there_are_not_styles() {
+		Monkey\Functions\when( 'beans_get_component_support' )->justReturn( true );
+		Monkey\Functions\when( 'get_option' )->justReturn( true );
+		Monkey\Functions\when( '_beans_is_compiler_dev_mode' )->justReturn( false );
+		Monkey\Functions\expect( 'add_query_arg' )->never();
+
+		// Check that beans_get() only gets called once.
+		Monkey\Functions\expect( 'beans_get' )->once()->andReturn( null );
+
+		// Check that this function does not get called.
+		Monkey\Functions\expect( 'beans_compile_css_fragments' )->never();
+
+		// Run the tests.
+		$this->assertNull( ( new _Beans_Page_Compiler() )->compile_page_styles() );
+	}
+
+	/**
+	 * Test _Beans_Page_Compiler::compile_page_styles() should not compile when assets are admin bar only.
+	 */
+	public function test_should_not_compile_when_assets_are_admin_bar_only() {
+		Monkey\Functions\when( 'beans_get_component_support' )->justReturn( true );
+		Monkey\Functions\when( 'get_option' )->justReturn( true );
+		Monkey\Functions\when( '_beans_is_compiler_dev_mode' )->justReturn( false );
+		Monkey\Functions\expect( 'add_query_arg' )->never();
+
+		// Initialize the mocked assets.
+		$assets = $this->get_mock_assets( [ 'admin-bar', 'open-sans', 'dashicons' ] );
+
+		// Check that beans_get() only gets called once.
+		Monkey\Functions\expect( 'beans_get' )->once()->andReturn( $assets );
+
+		// Check that beans_compile_css_fragments() does not get called.
+		Monkey\Functions\expect( 'beans_compile_css_fragments' )->never();
+
+		// Run the tests.
+		$this->assertNull( ( new _Beans_Page_Compiler() )->compile_page_styles() );
+	}
+
+	/**
+	 * Test _Beans_Page_Compiler::compile_page_styles() should not compile when the style is not registered.
+	 */
+	public function test_should_not_compile_when_style_is_not_registered() {
+		Monkey\Functions\when( 'beans_get_component_support' )->justReturn( true );
+		Monkey\Functions\when( 'get_option' )->justReturn( true );
+		Monkey\Functions\when( '_beans_is_compiler_dev_mode' )->justReturn( false );
+		Monkey\Functions\expect( 'add_query_arg' )->never();
+
+		// Initialize the mocked assets.
+		$assets                            = $this->get_mock_assets( [ 'admin-bar', 'uikit', 'child-style' ] );
+		$assets->registered['uikit']       = null;
+		$assets->registered['child-style'] = null;
+
+		// Set up the order of how beans_get() will be called.
```

Comment:
Please see previous comments about `Set up the order`
(I won't add more comments about it in this file)